### ISSUE TITLE
FAMD does not work

### ISSUE THREAD

**Issue by r0f1** [MaxHalford/prince#26]
X = pd.DataFrame(
data=[
["A", "A", "A", 2, 5, 7, 6, 3, 6, 7],
["A", "A", "A", 4, 4, 4, 2, 4, 4, 3],
["B", "A", "B", 5, 2, 1, 1, 7, 1, 1],
["B", "A", "B", 7, 2, 1, 2, 2, 2, 2],
["B", "B", "B", 3, 5, 6, 5, 2, 6, 6],
["B", "B", "A", 3, 5, 4, 5, 1, 7, 5]
])
famd = FAMD(
    n_components=3,
    n_iter=3,
    copy=True,
    engine='auto',
    random_state=4
)
famd = famd.fit(X)
famd_result = famd.transform(X)

raises

ValueError: could not convert string to float: 'A'

**Comment by ttpostle** [comment]
Can you provide the complete error message? I think it's supposed to Encode the categorical values as numbers first. My novice guess would be that it didn't correctly convert 'A' to an encoded number. I don't think it should be attempting to convert a Letter to a Float

**Comment by r0f1** [comment]
The full error message is:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-82e418407e17> in <module>()
     16 )
     17 famd = famd.fit(X)
---> 18 famd_result = famd.transform(X)

c:\users\frohrer\appdata\local\programs\python\python36\lib\site-packages\prince\pca.py in transform(self, X)
     79         """
     80         utils.validation.check_is_fitted(self, 's_')
---> 81         utils.check_array(X)
     82         return self.row_coordinates(X)
     83 

c:\users\frohrer\appdata\local\programs\python\python36\lib\site-packages\sklearn\utils\validation.py in check_array(array, accept_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, warn_on_dtype, estimator)
    446         # make sure we actually converted to numeric:
    447         if dtype_numeric and array.dtype.kind == "O":
--> 448             array = array.astype(np.float64)
    449         if not allow_nd and array.ndim >= 3:
    450             raise ValueError("Found array with dim %d. %s expected <= 2."

ValueError: could not convert string to float: 'A'

```

**Comment by r0f1** [comment]
I have found a workaround though: Disabling one of the sklearn checks.

```
from sklearn import utils
class FAMD2(FAMD):  
    def transform(self, X):
        utils.validation.check_is_fitted(self, 's_')
        return self.row_coordinates(X)

famd = FAMD2(
    n_components=3,
    n_iter=3,
    copy=True,
    engine='auto',
    random_state=4
)
famd = famd.fit(X)
famd_result = famd.transform(X)
famd_result.head()

```
Best, 
Flo

**Comment by MaxHalford** [comment]
This is now fixed with version 0.4.3.

**Comment by r0f1** [comment]
Appreciate it! :)
